### PR TITLE
[#9219] live server: NPE in feedbackResponseAdjustment when changing student data

### LINKED ISSUE #9219: live server: NPE in feedbackResponseAdjustment when changing student data
This kind of error reports have been coming from the live server for a long time. Previously ignored as not a high priority

>Error Message: teammates.common.util.Logger severe: Error encountered while adjusting feedback session responses of Paper Discussion 2 in course ANT3333: null
TEAMMATESLOG|||feedbackResponseAdjustment|||Servlet Action Failure|||true|||Unknown|||Unknown|||Unknown|||Unknown|||Servlet Action failure in feedbackResponseAdjustment
class java.lang.NullPointerException: 
java.lang.NullPointerException
at teammates.logic.core.FeedbackResponsesLogic.updateFeedbackResponseForChangingSection(FeedbackResponsesLogic.java:608)
at teammates.logic.core.StudentsLogic.adjustFeedbackResponseForEnrollments(StudentsLogic.java:507)
at teammates.logic.api.Logic.adjustFeedbackResponseForEnrollments(Logic.java:2175)
at teammates.ui.automated.FeedbackResponseAdjustmentWorkerAction.execute(FeedbackResponseAdjustmentWorkerAction.java:66)
at teammates.ui.automated.AutomatedServlet.doPost(AutomatedServlet.java:46)
at javax.servlet.http.HttpServlet.service(HttpServlet.java:707)
at javax.servlet.http.HttpServlet.service(HttpServlet.java:790)
at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:848)
at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1772)
at com.googlecode.objectify.ObjectifyFilter.doFilter(ObjectifyFilter.java:48)
at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1759)
at com.google.apphosting.utils.servlet.JdbcMySqlConnectionCleanupFilter.doFilter(JdbcMySqlConnectionCleanupFilter.java:60)
at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1759)
at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:582)
at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143)
at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:524)
at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:226)
at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143)
at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:134)
at com.google.apphosting.runtime.jetty9.ParseBlobUploadHandler.handle(ParseBlobUploadHandler.java:119)
at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1182)
at com.google.apphosting.runtime.jetty9.AppEngineWebAppContext.doHandle(AppEngineWebAppContext.java:171)
at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:512)
at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:185)
at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1112)
at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)
at com.google.apphosting.runtime.jetty9.AppVersionHandlerMap.handle(AppVersionHandlerMap.java:296)
at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:134)
at org.eclipse.jetty.server.Server.handle(Server.java:539)
at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:333)
at com.google.apphosting.runtime.jetty9.RpcConnection.handle(RpcConnection.java:202)
at com.google.apphosting.runtime.jetty9.RpcConnector.serviceRequest(RpcConnector.java:81)
at com.google.apphosting.runtime.jetty9.JettyServletEngineAdapter.serviceRequest(JettyServletEngineAdapter.java:123)
at com.google.apphosting.runtime.JavaRuntime$RequestRunnable.dispatchServletRequest(JavaRuntime.java:690)
at com.google.apphosting.runtime.JavaRuntime$RequestRunnable.dispatchRequest(JavaRuntime.java:652)
at com.google.apphosting.runtime.JavaRuntime$RequestRunnable.run(JavaRuntime.java:622)
at com.google.apphosting.runtime.JavaRuntime$NullSandboxRequestRunnable.run(JavaRuntime.java:816)
at com.google.apphosting.runtime.ThreadGroupPool$PoolEntry.run(ThreadGroupPool.java:273)
at java.lang.Thread.run(Thread.java:745)

### PR DESCRIPTION
**Pull request by junming403**
Fixes #9219 

Push to release branch to testify hypothesis about the cause of NPE.

### CODE REVIEW

**Comment by junming403** [comment]
ready for review

File: src/main/java/teammates/logic/core/FeedbackResponsesLogic.java [hunk 1]
```diff
@@ -605,6 +605,12 @@ public void updateFeedbackResponseForChangingSection(
             FeedbackResponseAttributes response) throws InvalidParametersException, EntityDoesNotExistException {
 
         FeedbackResponse feedbackResponse = frDb.getFeedbackResponseEntityOptimized(response);
+        if (feedbackResponse == null) {
+            // the response might have already being deleted.
```
Comment (xpdavid) [inline src/main/java/teammates/logic/core/FeedbackResponsesLogic.java]: due to?

Comment (junming403) [reply src/main/java/teammates/logic/core/FeedbackResponsesLogic.java]: Updated comments. Is it okay to state it in this way?

**Comment by damithc** [comment]
> LGTM! @damithc Do we want to deploy it to live soon?

Yes, we can do that.

**Comment by damithc** [comment]
@junming403 PR name should match issue name

**Comment by junming403** [comment]
PR name corrected.

**Pull request merged**